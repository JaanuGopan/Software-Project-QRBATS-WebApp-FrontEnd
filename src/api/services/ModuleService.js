import axios from "axios";
import ApiConstants from "../config/ApiConstants";

class ModuleService {
  static getModulesByUserId = async (userId) => {
    try {
      const response = await axios.get(
        ApiConstants.getAllModulesByUserIdUrl + `/${userId}`
      );
      return response.data;
    } catch (error) {
      console.log("Fail to get modules. " + error);
    }
  };

  static createModule = async (
    moduleCode,
    moduleName,
    moduleEnrolmentKey,
    semester,
    departmentId,
    userId
  ) => {
    try {
      const response = await axios.post(ApiConstants.createModuleUrl, {
        moduleCode: moduleCode,
        moduleName: moduleName,
        moduleEnrolmentKey: moduleEnrolmentKey,
        semester: semester,
        departmentId: departmentId,
        userId: userId,
      });
      return response;
    } catch (error) {
      console.log("Fail to create modules. " + error);
      if (error.response.status === 400) {
        console.log("Error Response is ", error.response);
        return error.response;
      }
    }
  };

  static updateModule = async (
    moduleId,
    moduleCode,
    moduleName,
    moduleEnrolmentKey,
    semester,
    departmentId,
    userId
  ) => {
    try {
      const response = await axios.put(ApiConstants.updateModuleUrl, {
        moduleId: moduleId,
        moduleCode: moduleCode,
        moduleName: moduleName,
        moduleEnrolmentKey: moduleEnrolmentKey,
        semester: semester,
        departmentId: departmentId,
        lectureId: userId,
      });
      return response;
    } catch (error) {
      console.error("Error In Updating Module.", error);
      if (error.response.status === 400) {
        console.log("Error Response is ", error.response);
        return error.response;
      }
    }
  };

  static deleteModuleById = async (moduleId) => {
    try {
      const response = await axios.delete(
        ApiConstants.deleteModuleUrl + "/" + `${moduleId}`
      );
      return response;
    } catch (error) {
      console.error("fail to delete module : " + error);
      return error.response.data;
    }
  };

  static getAllModulesByDepartmentId = async (departmentId) => {
    try {
      const response = await axios.get(
        ApiConstants.getAllModulesByDepartmentId + "/" + `${departmentId}`
      );
      if (response) {
        return response.data;
      }
    } catch (e) {
      console.log("Error in getting modules " + e);
    }
  };
}

export default ModuleService;
