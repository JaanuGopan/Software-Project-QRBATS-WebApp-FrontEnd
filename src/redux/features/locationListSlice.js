import { createSlice } from "@reduxjs/toolkit";

export const locationListSlice = createSlice({
  name: "locationList",
  initialState: ["NCC", "LT1", "LT2", "Auditorium", "DEIE", "DMME", "DCEE"],
  reducers: {
    setLocationList: (state, action) => {
      state = [];
      state.push(...state, action.payload);
    },
    resetLocationList: (state) => {
      state = [];
    },
  },
});

export const { setLocationList, resetLocationList } = locationListSlice.actions;
export default locationListSlice.reducer;
