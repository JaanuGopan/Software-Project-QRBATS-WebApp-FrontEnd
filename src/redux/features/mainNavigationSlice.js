import { createSlice } from "@reduxjs/toolkit";

export const mainNavigationSlice = createSlice({
  name: "mainNavigation",
  initialState: {
    sideBarIndex:
      localStorage.getItem("sideBarIndex") != null
        ? parseInt(localStorage.getItem("sideBarIndex"))
        : 0,
  },
  reducers: {
    setSideBarIndex: (state, action) => {
      state.sideBarIndex = action.payload;
      localStorage.setItem("sideBarIndex", action.payload);
    },
    resetSideBarIndex: (state) => {
      state.mainNavigation = 0;
      localStorage.removeItem("sideBarIndex");
    },
  },
});

export const { setSideBarIndex, resetSideBarIndex } =
  mainNavigationSlice.actions;
export default mainNavigationSlice.reducer;
