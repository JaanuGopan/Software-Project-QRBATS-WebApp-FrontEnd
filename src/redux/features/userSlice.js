import { createSlice } from '@reduxjs/toolkit';

export const userSlice = createSlice({
  name: 'user',
  initialState: {
    user: JSON.parse(localStorage.getItem('user')) || null,
    isLoggedIn: !!localStorage.getItem('authToken'),
    authToken: localStorage.getItem('authToken') || null,
  },
  reducers: {
    login: (state, action) => {
      state.user = action.payload;
    },
    logout: (state) => {
      state.user = null;
      localStorage.removeItem('user');
      localStorage.removeItem('authToken');
    },
  },
});

export const { login, logout } = userSlice.actions;
export const selectUser = (state) => state.user.user;
export default userSlice.reducer;
export const selectIsLoggedIn = (state) => state.user.isLoggedIn;
