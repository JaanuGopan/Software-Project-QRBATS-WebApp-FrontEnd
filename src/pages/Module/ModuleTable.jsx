import React, { useState, useEffect } from 'react';
import ModuleService from '../../api/services/ModuleService';
import '../AdminDashboard/AdminDashboard.css';
import { BiEdit } from 'react-icons/bi';
import Department from '../../utils/Department';

const ModuleTable = ({
  handleOpenCreateModuleWindow,
  handleOpenUpdateModuleWindow,
  searchModule,
  onModuleClick,
  modulesList,
  handleDeleteModule,
}) => {
  const [moduleList, setModuleList] = useState([]);
  const [selectedModule, setSelectedModule] = useState(null);

  const departmentList = Department.studentDepartmentList;

  useEffect(() => {
    setModuleList(modulesList);
  }, [modulesList]);

  const handleModuleClick = (module) => {
    setSelectedModule(module);
    onModuleClick(module);
  };

  const handleReloadModuleList = async (userId) => {
    ModuleService.getModulesByUserId(userId).then((modules) => {
      setModuleList(modules);
    });
  };

  return (
    <div className="tableDesign">
      <table className="event-report-tableArrangement">
        <thead>
          <tr>
            <th>No</th>
            <th className="expand">Module Name</th>
            <th>Module Code</th>
            <th>Semester</th>
            <th>Department</th>
            <th>Edit</th>
          </tr>
        </thead>
        {moduleList.filter(
          (module) =>
            module.moduleName.toLowerCase().includes(searchModule.toLowerCase()) ||
            module.moduleCode.toLowerCase().includes(searchModule.toLowerCase())
        ).length > 0 ? (
          <tbody>
            {moduleList
              .filter(
                (module) =>
                  module.moduleName.toLowerCase().includes(searchModule.toLowerCase()) ||
                  module.moduleCode.toLowerCase().includes(searchModule.toLowerCase())
              )
              .map((module, index) => (
                <tr
                  key={index}
                  onClick={() => handleModuleClick(module)}
                  className={selectedModule === module ? 'selected-row' : 'event-row'}
                >
                  <td>{index + 1}</td>
                  <td>{module.moduleName}</td>
                  <td>{module.moduleCode}</td>
                  <td>{module.semester}</td>
                  <td>{departmentList[module.departmentId - 1]}</td>
                  <td>
                    <button onClick={handleOpenUpdateModuleWindow} className="ViewButton">
                      <BiEdit className="EditIcon" />
                      <p className="ViewButtonLabel">Edit</p>
                    </button>
                  </td>
                </tr>
              ))}
          </tbody>
        ) : (
          <tbody className="nodata">No Data Available</tbody>
        )}
      </table>
    </div>
  );
};

export default ModuleTable;
