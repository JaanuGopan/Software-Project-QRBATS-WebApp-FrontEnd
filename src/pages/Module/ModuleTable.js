import React, { useState, useEffect } from "react";
import { useSelector } from "react-redux";
import { selectUser } from "../../redux/features/userSlice";
import ModuleService from "../../api/services/ModuleService";
import { CiViewList } from "react-icons/ci";
import "../AdminDashboard/AdminDashboard.css";
import { BiEdit } from "react-icons/bi";

const ModuleTable = ({
  handleOpenCreateModuleWindow,
  handleOpenUpdateModuleWindow,
  searchModule,
  onModuleClick,
  modulesList,
  handleDeleteModule,
}) => {
  const user = useSelector(selectUser);
  const { userId } = user || {};
  const [moduleList, setModuleList] = useState([]);
  const [selectedModule, setSelectedModule] = useState(null);

  const departmentList = ["DEIE", "DCOM", "DMME", "DCEE", "DMENA"];

  useEffect(() => {
    setModuleList(modulesList);
  }, [modulesList]);

  const handleModuleClick = (module) => {
    setSelectedModule(module);
    onModuleClick(module);
  };

  const handleReloadModuleList = async (userId) => {
    ModuleService.getModulesByUserId(userId).then((modules) => {
      setModuleList(modules);
    });
  };

  return (
    <div className="tableDesign">
      <table className="event-report-tableArrangement">
        <thead>
          <tr>
            <th>No</th>
            <th className="expand">Module Name</th>
            <th>Module Code</th>
            <th>Semester</th>
            <th>Department</th>
            <th>Edit</th>
          </tr>
        </thead>
        <tbody>
          {moduleList
            .filter(
              (module) =>
                module.moduleName
                  .toLowerCase()
                  .includes(searchModule.toLowerCase()) ||
                module.moduleCode
                  .toLowerCase()
                  .includes(searchModule.toLowerCase())
            )
            .map((module, index) => (
              <tr
                key={index}
                onClick={() => handleModuleClick(module)}
                className={
                  selectedModule === module ? "selected-row" : "event-row"
                }
              >
                <td>{index + 1}</td>
                <td>{module.moduleName}</td>
                <td>{module.moduleCode}</td>
                <td>{module.semester}</td>
                <td>{departmentList[module.departmentId - 1]}</td>
                <td>
                  <button
                    onClick={handleOpenUpdateModuleWindow}
                    className="ViewButton"
                  >
                    <BiEdit className="EditIcon" />
                    <p className="ViewButtonLabel">Edit</p>
                  </button>
                </td>
              </tr>
            ))}
        </tbody>
      </table>
    </div>
  );
};

export default ModuleTable;
