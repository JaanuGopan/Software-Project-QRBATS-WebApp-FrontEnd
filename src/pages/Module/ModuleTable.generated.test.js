import renderer from 'react-test-renderer';
import React, { useState, useEffect } from "react";
import { useSelector } from "react-redux";
import { selectUser } from "../../redux/features/userSlice";
import ModuleService from "../../api/services/ModuleService";
import { CiViewList } from "react-icons/ci";

import { BiEdit } from "react-icons/bi";
import ModuleTable from "./ModuleTable";

jest.mock("react-redux");
jest.mock("../../redux/features/userSlice");
jest.mock("../../api/services/ModuleService");
jest.mock("react-icons/ci");
jest.mock("../AdminDashboard/AdminDashboard.css");
jest.mock("react-icons/bi");

const renderTree = tree => renderer.create(tree);
describe('<ModuleTable>', () => {
  it('should render component', () => {
    expect(renderTree(<ModuleTable 
    />).toJSON()).toMatchSnapshot();
  });
  
});