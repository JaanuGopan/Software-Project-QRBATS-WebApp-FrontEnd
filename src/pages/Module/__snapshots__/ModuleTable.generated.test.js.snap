// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<ModuleTable> should render component 1`] = `
<div
  className="tableDesign"
>
  <table
    className="event-report-tableArrangement"
  >
    <thead>
      <tr>
        <th>
          No
        </th>
        <th
          className="expand"
        >
          Module Name
        </th>
        <th>
          Module Code
        </th>
        <th>
          Semester
        </th>
        <th>
          Department
        </th>
        <th>
          Edit
        </th>
      </tr>
    </thead>
    <tbody />
  </table>
</div>
`;
